src/geodedic.rs

```diff
@@ -203,12 +226,35 @@ impl Coordinate<Ecef> {
 
     /// Converts an Earth-Centered, Earth-Fixed coordinate into latitude, longitude, and altitude.
     ///
-    /// Note that this conversion is not trivial and needs to be approximated. This implementation
-    /// currentaly uses [Ferrari's solution][ferrari], but this may change in the future.
+    /// Note that this conversion is not trivial and needs to be approximated.
+    ///
+    /// The implementation currently only guarantees conversion to Wgs84 datums with altitude
+    /// between -10km and 50km from the surface of the Wgs84 ellipsoid, roughly corresponding
+    /// to the bottom of the Mariana Trench to the top of the stratosphere. Outside this range,
+    /// the implementation may panic.
+    ///
+    /// This implementation currently uses [Ferrari's solution][ferrari], but this may change
+    /// in the future.
     ///
     /// [ferrari]: https://en.wikipedia.org/wiki/Geographic_coordinate_conversion#The_application_of_Ferrari's_solution
     #[must_use]
     pub fn to_wgs84(&self) -> Wgs84 {
+        let geo_center_distance_sq =
+            self.point.x * self.point.x + self.point.y * self.point.y + self.point.z * self.point.z;
+
+        if !(ECEF_TO_WGS84_MIN_GEO_CENTER_DISTANCE_M_SQ
+            ..=ECEF_TO_WGS84_MAX_GEO_CENTER_DISTANCE_M_SQ)
+            .contains(&geo_center_distance_sq)
+        {
+            if geo_center_distance_sq < f64::EPSILON {
+                panic!("conversion from Ecef to Wgs84 at coordinate origin is not supported");
+            } else {
+                panic!(
+                    "conversion from Ecef to Wgs84 outside altitude range {ECEF_TO_WGS84_MIN_ALTITUDE_M}..{ECEF_TO_WGS84_MAX_ALTITUDE_M} is not supported: {self}",
```

Comment:
```suggestion
                    "conversion from ECEF to WGS84 outside altitude range {ECEF_TO_WGS84_MIN_ALTITUDE_M}..{ECEF_TO_WGS84_MAX_ALTITUDE_M} is not supported: {self}",
```
Would also probably break this into two lines with `\`, otherwise I suspect rustfmt will be sad.